lib/ansible/modules/remote_mount_facts.py

```diff
@@ -0,0 +1,290 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2024 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import annotations
+
+
+DOCUMENTATION = """
+---
+module: remote_mount_facts
+version_added: 2.18
+short_description: Get mount facts for devices that do not start with a path separator.
+description:
+  - Get mount facts for devices that do not start with a path separator, which are skipped by default mount gathering.
+options:
+  collect:
+    description: A list of mount types to return.
+    type: list
+    elements: str
+    required: true
+    choices:
+      - cifs
+      - ceph
+      - fuse
+      - glusterfs
+      - gpfs
+      - nfs
+      - nfs4
+      - vboxsf
+      - zfs
+  timeout:
+    description: The maximum number of seconds to query for each mount type in O(collect).
+    default: 20
```

Comment:
maybe 0/none should be the default, most users of this module should not have too many issues and this should only be set if they do encounter them.

Reply:
I made the module default 0, which internally times out after .01 seconds.

Reply:
why i was thinking None was better default, only pass/call timeout when `is not None`, can do same with 0 but I think `None` makes a better sentinel here.

Reply:
Only calling timeout when timeout `is not None/0` allows a hang, which means anyone enabling this module for gather_facts will also need to set the timeout. I think it should error after a short period of time by default instead (which is what I've done).

Reply:
ok, i was just thinking most users won't have hangs but can have long waits for network items, since this is an explicit choice we could default to waiting, unlike 'default fact gathering' in which waiting can become too cumbersome.

Reply:
I guess I misunderstood the request. I thought 20 seconds was erring on an unreasonable amount of time to wait since it's per mount, but could make it longer too ¯\_(ツ)_/¯ Making `timeout(seconds)` conditional introduces what I'd consider to be a bug.

Reply:
>  ... unlike 'default fact gathering' in which waiting can become too cumbersome.

I am specifically thinking of people who are going to enable this as a FACTS_MODULE.

Reply:
I circled back around to this, and fixed using no timeout, thanks for pointing it out.